lib/aws-aurora-serverless-stack.ts

```diff
@@ -1,8 +1,83 @@
 import * as cdk from 'aws-cdk-lib';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as rds from 'aws-cdk-lib/aws-rds';
+import * as kms from 'aws-cdk-lib/aws-kms';
 import { Construct } from 'constructs';
+import { AuroraEngine, AwsAuroraServerlessStackProps } from './AwsAuroraServerlessStackProps';
+import { SecretValue } from 'aws-cdk-lib';
+import { parseVpcSubnetType } from '../utils/vpc-type-parser';
 
 export class AwsAuroraServerlessStack extends cdk.Stack {
-  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+  constructor(scope: Construct, id: string, props: AwsAuroraServerlessStackProps) {
     super(scope, id, props);
+
+    const vpc = ec2.Vpc.fromLookup(this, `${props.resourcePrefix}-VPC-Imported`, {
+      vpcId: props.vpcId,
+    });
+
+    const kmsKey = new kms.Key(this, `${props.resourcePrefix}-Aurora-KMS-Key`, {
+      enableKeyRotation: true,
+    });
+
+    const auroraPort = props.auroraEngine === AuroraEngine.AuroraPostgresql ? 5432 : 3306;
+    const auroraSecurityGroup = new ec2.SecurityGroup(this, `${props.resourcePrefix}-Aurora-Security-Group`, {
+      vpc,
+      allowAllOutbound: true,
+    });
+
+    const auroraDatabaseCluster = new rds.DatabaseCluster(this, `${props.resourcePrefix}-Aurora-Serverless`, {
+      engine: props.auroraEngine === AuroraEngine.AuroraPostgresql ?
+        rds.DatabaseClusterEngine.auroraPostgres({ version: rds.AuroraPostgresEngineVersion.VER_17_2 }) :
+        rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_08_0 }),
+      vpc,
+      vpcSubnets: {
+        subnetType: parseVpcSubnetType(props.vpcSubnetType),
+      },
+      securityGroups: [auroraSecurityGroup],
+      autoMinorVersionUpgrade: true,
+      serverlessV2MaxCapacity: props.serverlessV2MaxCapacity,
+      serverlessV2MinCapacity: props.serverlessV2MinCapacity,
+      writer: rds.ClusterInstance.serverlessV2('writer'),
+      readers: [rds.ClusterInstance.serverlessV2('reader')],
+      storageEncrypted: true,
+      storageEncryptionKey: kmsKey,
+      credentials: rds.Credentials.fromPassword(props.rdsUsername, SecretValue.unsafePlainText(props.rdsPassword)),
```

Comment:
**Suggestion:** The current implementation uses `SecretValue.unsafePlainText()` for database credentials, which is not recommended for production use. Instead, use AWS Secrets Manager to securely store and manage database credentials. This improves security by avoiding hardcoded credentials and enables credential rotation.
> 🎯  **High Priority** 
 >[Category: **`Security`**, Importance: **`10`**]
> **Reason:** This suggestion addresses a critical security issue by replacing plaintext database credentials with AWS Secrets Manager, which is a best practice for secure credential management. It significantly enhances the security of the application by enabling secure storage and rotation of credentials.

```suggestion
      credentials: rds.Credentials.fromSecret(new secretsmanager.Secret(this, `${props.resourcePrefix}-db-credentials`, {
        secretStringValue: SecretValue.jsonField({
          username: props.rdsUsername,
          password: props.rdsPassword
        }),
        encryptionKey: kmsKey
      })),
```
- [ ]  **I have reviewed this code suggestion and addressed the relevant ones.**